### PR TITLE
Tidy up build and leverage new slamdata commons dependencies

### PR DESCRIPTION
**Pull request by wemrysi**
Makes use of the new `slamdata-sbt` and predef and generally brings the build structure up to date.

When merging, just use the GH merge button for this PR as the previous version included a "SNAPSHOT" suffix which I'm not sure the merge script can handle, we should be able to use it going forward.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,13 @@
 language: scala
 scala:
-  - 2.11.7
+  - 2.11.8
+  - 2.12.1
 jdk:
-  - oraclejdk7
+  - oraclejdk8
```
Comment (sellout) [inline .travis.yml]: I think you should still build on 7 as well, and exclude the 2.12/jdk7 combo.

File: core/src/main/scala/abt/Arity.scala [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package abt
 
-import scala.collection.immutable.Vector
+import slamdata.Predef.Vector
```
Comment (sellout) [inline core/src/main/scala/abt/Arity.scala]: Should just use `slamdata.Predef._`.

File: core/src/main/scala/abt/Coord.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package abt
+
+import slamdata.Predef.Int
```
Comment (sellout) [inline core/src/main/scala/abt/Coord.scala]: Same.

File: core/src/main/scala/abt/View.scala [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package abt
 
-import scala.collection.immutable.Vector
+import slamdata.Predef.Vector
```
Comment (sellout) [inline core/src/main/scala/abt/View.scala]: Same.

**Review by sellout — CHANGES_REQUESTED** [review]
Very excited to see this updated.

**Comment by wemrysi** [comment]
@sellout Thanks for the feedback, updated.

**Review by sellout — APPROVED** [review]

**Review by drostron — APPROVED** [review]
👍🏾

**Pull request merged**